src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -209,5 +216,2406 @@ public async Task DirectoryEntryWithData(bool async)
+
+            Assert.Throws<InvalidDataException>(() =>
+            {
+                entry.ExtractToFile(tempFile, overwrite: true); // No password passed
+            });
+        }
+
+
+        [Fact]
+        public async Task ExtractToFileAsync_WithPassword_ShouldCreatePlaintextFile()
+        {
+            string zipPath = @"C:\Users\spahontu\Downloads\test.zip";
+            Assert.True(File.Exists(zipPath), $"Test ZIP not found at {zipPath}");
+
+            string tempFile = Path.Combine(Path.GetTempPath(), "hello_async.txt");
+            if (File.Exists(tempFile)) File.Delete(tempFile);
+
+            using var archive = ZipFile.OpenRead(zipPath);
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("hello.txt", StringComparison.OrdinalIgnoreCase));
+
+            await entry.ExtractToFileAsync(tempFile, overwrite: true, password: "123456789");
+
+            Assert.True(File.Exists(tempFile), "Extracted file was not created.");
+            string content = await File.ReadAllTextAsync(tempFile);
+            Assert.Equal("Hello ZipCrypto!", content);
+
+            File.Delete(tempFile);
+        }
+
+        [Fact]
+        public async Task ExtractToFileAsync_WithWrongPassword_ShouldThrow()
+        {
+            string zipPath = @"C:\Users\spahontu\Downloads\test.zip";
+            Assert.True(File.Exists(zipPath), $"Test ZIP not found at {zipPath}");
+
+            string tempFile = Path.Combine(Path.GetTempPath(), "hello_async.txt");
+            if (File.Exists(tempFile)) File.Delete(tempFile);
+
+            using var archive = ZipFile.OpenRead(zipPath);
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("hello.txt", StringComparison.OrdinalIgnoreCase));
+
+            await Assert.ThrowsAsync<InvalidDataException>(async () =>
+            {
+                await entry.ExtractToFileAsync(tempFile, overwrite: true, password: "wrongpass");
+            });
+        }
+
+
+        [Fact]
+        public async Task ExtractToFileAsync_WithoutPassword_ShouldThrow()
+        {
+            string zipPath = @"C:\Users\spahontu\Downloads\test.zip";
+            Assert.True(File.Exists(zipPath), $"Test ZIP not found at {zipPath}");
+
+            string tempFile = Path.Combine(Path.GetTempPath(), "hello_async.txt");
+            if (File.Exists(tempFile)) File.Delete(tempFile);
+
+            using var archive = ZipFile.OpenRead(zipPath);
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("hello.txt", StringComparison.OrdinalIgnoreCase));
+
+            await Assert.ThrowsAsync<InvalidDataException>(async () =>
+            {
+                await entry.ExtractToFileAsync(tempFile, overwrite: true, cancellationToken: default); // No password passed
+            });
+        }
+
+        [Fact]
+        public async Task ExtractToFileAsync_WithCancellation_ShouldCancel()
```

Comment:
I would expect there to be a similar test already, so maybe we can just augment it to test the overload which takes the password